src/features/metadata-based-view/__tests__/IconWithTooltip.test.js

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+
+import { IconWithTooltipComponent as IconWithTooltip } from '../IconWithTooltip';
+
+import messages from '../../../elements/common/messages';
+import { CANCEL_ICON_TYPE, EDIT_ICON_TYPE, SAVE_ICON_TYPE } from '../constants';
+
+describe('features/metadata-based-view/IconWithTooltip', () => {
+    const getWrapper = props => mount(<IconWithTooltip {...props} />);
+    const onClick = () => {};
+
+    test('Get IconClose with Tooltip to cancel edited value', () => {
+        const props = {
+            className: 'bdl-MetadataBasedItemList-cell--cancelIcon',
+            onClick,
+            tooltipText: messages.cancel,
+            type: CANCEL_ICON_TYPE,
+        };
+        const wrapper = getWrapper(props);
+        expect(wrapper).toMatchSnapshot();
```

Comment:
Are there specific assertions you want to make about the mounted (or shallow mounted) component? Seems like snapshots generally get ignored (especially fully mounted ones)

Reply:
Because this is a stateless functional component, there are no lifecycle methods and their behaviors to test. The snapshot testing in this case makes sense to me.